File: benchmarks/tsb/bench_cat_from_codes.ts
/**
 * Benchmark: catFromCodes on 100k-element code array
 */
import { catFromCodes } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const categories = ["alpha", "beta", "gamma", "delta"];
const codes = Array.from({ length: ROWS }, (_, i) => i % 4);

for (let i = 0; i < WARMUP; i++) catFromCodes(codes, categories);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) catFromCodes(codes, categories);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "cat_from_codes",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_cat_from_codes.py
"""Benchmark: Categorical from codes on 100k-element array"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

categories = ["apple", "banana", "cherry", "date", "elderberry"]
codes = np.arange(ROWS) % len(categories)

for _ in range(WARMUP):
    pd.Categorical.from_codes(codes, categories=categories)

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.Categorical.from_codes(codes, categories=categories)
total = (time.perf_counter() - start) * 1000

print(json.dumps({ "function": "cat_from_codes", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))
